dev/NavigationView/NavigationView.cpp

```diff
@@ -1381,19 +1652,58 @@ void NavigationView::AnimateSelectionChanged(const winrt::IInspectable& prevItem
             ResetElementAnimationProperties(nextIndicator, 1.0f);
         }
 
-        if (m_lastSelectedItemPendingAnimationInTopNav.get())
-        {
-            // if nextItem && !nextIndicator, that means a item from topnav flyout is selected, and we delay the animation to LayoutUpdated.
-            // nextIndicator is null because we have problem to get the selectionindicator since it's not in primary list yet.
-            // Otherwise we already done the animation and clear m_lastSelectedItemPendingAnimationInTopNav.
-            if (!(nextItem && !nextIndicator))
-            {
-                m_lastSelectedItemPendingAnimationInTopNav.set(nullptr);
-            }
-        }
+        m_activeIndicator.set(nextIndicator);
     }
 }
 
+void NavigationView::PlayIndicatorNonSameLevelAnimations(const winrt::UIElement& indicator, bool isOutgoing, bool fromTop)
+{
+    winrt::Visual visual = winrt::ElementCompositionPreview::GetElementVisual(indicator);
```

Comment:
>winrt::Visual visual = winrt::ElementCompositionPreview::GetElementVisual(indicator); [](start = 4, length = 85)

Don't use ElementCompositionPreview, it is incompatable with a buch of APIs, instead use the UIElement's facade properties.  Unfortunately those aren't available downlevel so you probably need to keep this version for rs3 (and maybe 4).

Reply:
Hmm, looking closer there is ECP all over this control... maybe we should just make a task to clean it up... but this is pretty broken.

---
In reply to: [393872652](https://github.com/microsoft/microsoft-ui-xaml/pull/2004#discussion_r393872652) [](ancestors = 393872652)

Reply:
@ojhad Please create an issue and link it here.

Reply:
https://github.com/microsoft/microsoft-ui-xaml/issues/2174